specification/hybridconnectivity/HybridConnectivity.Management/EndpointResource.tsp

```diff
@@ -0,0 +1,167 @@
+import "@typespec/rest";
+import "./models.tsp";
+
+using TypeSpec.Rest;
+using Azure.ResourceManager;
+using TypeSpec.Http;
+using TypeSpec.OpenAPI;
+
+namespace Microsoft.HybridConnectivity;
+/**
+ * The endpoint for the target resource.
+ */
+model EndpointResource
+  is Azure.ResourceManager.ExtensionResource<EndpointProperties> {
+  ...ResourceNameParameter<
+    Resource = EndpointResource,
+    KeyName = "endpointName",
+    SegmentName = "endpoints",
+    NamePattern = ""
+  >;
+}
+
+@armResourceOperations
+interface EndpointResources {
+  /**
+   * Gets the endpoint to the resource.
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-openapi" "non-standard operations"
+  @operationId("Endpoints_Get")
+  get is ArmResourceRead<
+    EndpointResource,
+    Azure.ResourceManager.Foundations.ExtensionBaseParameters
+  >;
+
+  /**
+   * Create or update the endpoint to the target resource.
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-openapi" "non-standard operations"
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-put-operation-response-codes"
+  @operationId("Endpoints_CreateOrUpdate")
+  createOrUpdate is ArmResourceCreateOrReplaceSync<
+    EndpointResource,
+    Azure.ResourceManager.Foundations.ExtensionBaseParameters,
+    Response = ArmResourceUpdatedResponse<EndpointResource>
+  >;
+
+  /**
+   * Update the endpoint to the target resource.
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-openapi" "non-standard operations"
+  @parameterVisibility
+  @operationId("Endpoints_Update")
+  update is ArmCustomPatchSync<
+    EndpointResource,
+    EndpointResource,
+    Azure.ResourceManager.Foundations.ExtensionBaseParameters
+  >;
+
+  /**
+   * Deletes the endpoint access to the target resource.
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-openapi" "non-standard operations"
+  #suppress "@azure-tools/typespec-azure-core/no-response-body" "For backward compatibility"
+  @operationId("Endpoints_Delete")
+  delete is ArmResourceDeleteSync<
+    EndpointResource,
+    Azure.ResourceManager.Foundations.ExtensionBaseParameters
+  >;
+
+  /**
+   * List of endpoints to the target resource.
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-openapi" "non-standard operations"
+  @operationId("Endpoints_List")
+  list is ArmResourceListByParent<
+    EndpointResource,
+    Azure.ResourceManager.Foundations.ExtensionBaseParameters,
+    Response = EndpointsList
+  >;
+
+  /**
+   * Gets the endpoint access credentials to the resource.
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-openapi" "non-standard operations"
+  #suppress "@azure-tools/typespec-azure-core/casing-style" "The name of the body starts with a capital letter"
+  @operationId("Endpoints_ListCredentials")
+  listCredentials is ArmResourceActionSync<
+    EndpointResource,
+    {
+      /**
+       * Object of type ListCredentialsRequest
+       */
+      @bodyRoot ListCredentialsRequest?: ListCredentialsRequest;
```

Comment:
`@body` should be better since template has already had `@bodyRoot`: https://github.com/Azure/typespec-azure/blob/b0da6dce6c56ed5ee0a711f3902ab52a4f202467/packages/typespec-azure-resource-manager/lib/operations.tsp#L566

Reply:
+1 the same issue on operation `listIngressGatewayCredentials`.

Reply:
I was wondering for this operation if the request body parameter name and description matters, if not we could update it to 
```
listCredentials is ArmResourceActionSync<
    EndpointResource,
    ListCredentialsRequest,
    ....
```
If yes we could update as following
```
listCredentials is ArmResourceActionSync<
    EndpointResource,
    /**
       * Object of type ListCredentialsRequest
       */
    @body listCredentialsRequest?: ListCredentialsRequest;,
    ....
```

Reply:
We write this way because the body in the original swagger is optional, so that we cannot pass in the request body directly. 

We could change to `@body` as you wrote above.